### PR TITLE
Save scene and global checkbox options of History window

### LINKED ISSUE #88432: History options do not save your status.
### Tested versions

v4.3.dev3.official [36e943b6b]

### System information

w10 64

### Issue description

Override all history options.
Reload the project and the options are still activated.

https://github.com/godotengine/godot/assets/19652075/b2fd6eb6-63a7-4f7e-925f-74ae77224dab

### Steps to reproduce

Watch the video

### Minimal reproduction project (MRP)

...

### PR DESCRIPTION
**Pull request by emre0altan**
See the title please
* *Fixes: #88432*

### CODE REVIEW

File: editor/history_dock.cpp [hunk 1]
```diff
@@ -54,6 +55,8 @@ void HistoryDock::refresh_history() {
 	action_list->clear();
 	bool include_scene = current_scene_checkbox->is_pressed();
 	bool include_global = global_history_checkbox->is_pressed();
+	EditorSettings::get_singleton()->set_project_metadata("history", "include_scene", include_scene);
+	EditorSettings::get_singleton()->set_project_metadata("history", "include_global", include_global);
```
Comment (KoBeWi) [inline editor/history_dock.cpp:59]: This shouldn't be done on every history refresh. You'll need to add a new method to connect to the `toggled` signal.

Comment (emre0altan) [reply editor/history_dock.cpp:59]: Wouldn't it be the same functionality because `refresh_history` is connected to `toggled` signal?

Comment (KoBeWi) [reply editor/history_dock.cpp:59]: `refresh_history()` is also called on undoredo action.

Comment (emre0altan) [reply editor/history_dock.cpp:59]: Oh, okay. I'll update it now, thanks.

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**